packages/api/Tests/UnitTests/Vacation/VacationTests.cs

```diff
@@ -0,0 +1,127 @@
+﻿using AlvTime.Business.Options;
+using AlvTime.Business.TimeRegistration;
+using AlvTime.Business.Users;
+using AlvTime.Business.Utils;
+using AlvTime.Persistence.DatabaseModels;
+using AlvTime.Persistence.Repositories;
+using Microsoft.Extensions.Options;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace Tests.UnitTests.Vacation
+{
+    public class VacationTests
+    {
+        private readonly AlvTime_dbContext _context;
+        private readonly IOptionsMonitor<TimeEntryOptions> _options;
+        private readonly Mock<IUserContext> _userContextMock;
+        private readonly TimeRegistrationService _timeRegistrationService;
+
+        public VacationTests()
+        {
+            _context = new AlvTimeDbContextBuilder()
+                .WithTasks()
+                .WithLeaveTasks()
+                .WithProjects()
+                .WithUsers()
+                .WithCustomers()
+                .CreateDbContext();
+
+            var entryOptions = new TimeEntryOptions
+            {
+                SickDaysTask = 14,
+                PaidHolidayTask = 13,
+                UnpaidHolidayTask = 19,
+                FlexTask = 18,
+                StartOfOvertimeSystem = new DateTime(2020, 01, 01),
+                AbsenceProject = 9
+            };
+
+            _options = Mock.Of<IOptionsMonitor<TimeEntryOptions>>(options => options.CurrentValue == entryOptions);
+            _userContextMock = new Mock<IUserContext>();
+
+            var user = new AlvTime.Business.Users.User
+            {
+                Id = 1,
+                Email = "someone@alv.no",
+                Name = "Someone"
+            };
+
+            _userContextMock.Setup(context => context.GetCurrentUser()).Returns(System.Threading.Tasks.Task.FromResult(user));
+
+            _timeRegistrationService = CreateTimeRegistrationService();
+        }
+
+        [Fact]
+        public async System.Threading.Tasks.Task CorrectNumberOfVacationHours()
```

Comment:
Bra med masse tester! Ønsker meg kanskje en litt annen navning på testene, typ Funksjonalitet_Input_ForventetResultat. Så: RegisterVacation_CorrectHoursAmount_VacationRegistered for eksempel.